stl/inc/mdspan

```diff
@@ -24,151 +24,126 @@ _STL_DISABLE_CLANG_WARNINGS
+        array<rank_type, rank() + 1> _Result{};
+        rank_type _Counter = 0;
+        for (rank_type _Idx = 0; _Idx < rank(); ++_Idx) {
+            _Result[_Idx] = _Counter;
+            if (_Static_extents[_Idx] == dynamic_extent) {
                 ++_Counter;
             }
         }
+        _Result[rank()] = _Counter;
         return _Result;
+#pragma warning(pop) // TRANSITION, "/analyze:only" BUG?
     }
 
-    index_type _Dynamic_extents[_Rank_dynamic]                           = {};
-    static constexpr size_t _Static_extents[sizeof...(_Extents)]         = {_Extents...};
-    static constexpr array<size_t, sizeof...(_Extents)> _Dynamic_indices = _Get_dynamic_indices();
+    static constexpr array<rank_type, rank()> _Static_extents      = {_Extents...};
+    static constexpr array<rank_type, rank() + 1> _Dynamic_indices = _Make_dynamic_indices();
 
-    constexpr _Mdspan_extent_type() noexcept = default;
-
-    template <class... _OtherIndexTypes>
-        requires (sizeof...(_OtherIndexTypes) == _Rank_dynamic) && (is_same_v<_OtherIndexTypes, index_type> && ...)
-    constexpr _Mdspan_extent_type(_OtherIndexTypes... _OtherExtents) noexcept : _Dynamic_extents{_OtherExtents...} {}
-
-    template <class _OtherIndexType, size_t _Size, size_t... _Idx>
-        requires (_Size == _Rank_dynamic)
-    constexpr _Mdspan_extent_type(span<_OtherIndexType, _Size> _Data, index_sequence<_Idx...>) noexcept
-        : _Dynamic_extents{static_cast<index_type>(_STD as_const(_Data[_Idx]))...} {}
-
-    template <class... _OtherIndexTypes>
-        requires (sizeof...(_OtherIndexTypes) == sizeof...(_Extents)) && (sizeof...(_Extents) != _Rank_dynamic)
-              && (is_same_v<_OtherIndexTypes, index_type> && ...)
-    constexpr _Mdspan_extent_type(_OtherIndexTypes... _OtherExtents) noexcept {
-        auto _It = _Dynamic_extents;
-        ((_Extents == dynamic_extent ? void(*_It++ = _OtherExtents) : void(_OtherExtents)), ...);
+    _NODISCARD static constexpr rank_type _Dynamic_index(rank_type _Idx) noexcept {
+        return _Dynamic_indices[_Idx];
     }
 
-    template <class _OtherIndexType, size_t _Size, size_t... _Idx>
-        requires (_Size == sizeof...(_Extents)) && (sizeof...(_Extents) != _Rank_dynamic)
-    constexpr _Mdspan_extent_type(span<_OtherIndexType, _Size> _Data, index_sequence<_Idx...>) noexcept
-        : _Dynamic_extents{{static_cast<index_type>(_STD as_const(_Data[_Dynamic_indices[_Idx]]))...}} {}
-
-    constexpr index_type* _Begin_dynamic_extents() noexcept {
-        return _Dynamic_extents;
-    }
-
-    constexpr const index_type* _Begin_dynamic_extents() const noexcept {
-        return _Dynamic_extents;
+    _NODISCARD static _CONSTEVAL auto _Make_dynamic_indices_inv() noexcept {
+        array<rank_type, rank()> _Result{};
+        for (rank_type _Idx = 0; _Idx < rank(); ++_Idx) {
+            for (rank_type _Rdx = 0; _Rdx < rank(); ++_Rdx) {

```

Comment:
Suggestion: `_Idx` and `_Rdx` differ by a single character, making this somewhat difficult to read and mistake-prone. Could a clearer name be found for `_Rdx`?

Reply:
`_Idx_inv` might be better name (I'm going to include it in my next PR).